### PR TITLE
Csv reader

### LINKED ISSUE #16: Read locations from CSV file and show list of locations

### PR DESCRIPTION
**Pull request by gibrane**
fixed issues for M7 demo

Closes #16

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -25,6 +25,11 @@
             </intent-filter>
         </activity>
         <activity android:name=".controllers.ApplicationActivity" />
+        <activity android:name=".controllers.ListViewCustom" />
+        <activity
+            android:name=".controllers.ListElementPage"></activity>
+            <!-- android:label="@string/title_activity_list_element_page"-->
```
Comment (evan10s) [inline app/src/main/AndroidManifest.xml]:
```suggestion

```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import android.widget.ListView;
+import android.app.Activity;
+import android.widget.TextView;
+
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java]: Can you check this for unused imports and remove those?

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        Intent intent = getIntent();
+        setContentView(R.layout.csv_element_view);
+
+
+        tv = findViewById(R.id.csv_text);
+        tv.setText(String.format("%s%n%s%n%s%n%s%n%s%n%s%n",intent.getStringExtra("Name"), intent.getStringExtra("Address"), intent.getStringExtra("Phone Number"),
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java]: Make this better

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import android.widget.ListView;
+import android.app.Activity;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: Remove unused imports

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+public class ListViewCustom extends AppCompatActivity {
+    private List<ListElement> locationSamples = new ArrayList<>();
+    private ListView listView;
+//    private TextView csvElementView;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion

```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.csv_view);
+        readLocationData();
+
+//        csvElementView = findViewById(R.id.csv_text);
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion

```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+                intent.putExtra("Address", e.getStreetAddress());
+                intent.putExtra("Phone Number", e.getPhoneNumber());
+                intent.putExtra("Longitude", e.getLongitude());
+                intent.putExtra("Latitude", e.getLatitude());
+                intent.putExtra("Type", e.getLocationType());
+
+                Bundle bund = intent.getExtras();
+                startActivity(intent);
+
+
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+            }
+
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+        });
+    }
+
+    private void readLocationData() {
+
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+                listElement.setStreetAddress(tokens[4]);
+                listElement.setCity(tokens[5]);
+                listElement.setState(tokens[6]);
+                listElement.setZipCode(Integer.parseInt(tokens[7]));
+                listElement.setLocationType(tokens[8]);
+                listElement.setPhoneNumber(tokens[9]);
+                listElement.setWebsite(tokens[10]);
+                locationSamples.add(listElement);
+
+                Log.d("MainActivity", "Just Created: " + listElement);
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
```
Remove log statements

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+
+            }
+        } catch (IOException e) {
+            Log.e("MainActivity", "error reading assets", e);
+        }
+
+
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: remove whitespace

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+                listElement.setLocationType(tokens[8]);
+                listElement.setPhoneNumber(tokens[9]);
+                listElement.setWebsite(tokens[10]);
+                locationSamples.add(listElement);
+
+                Log.d("MainActivity", "Just Created: " + listElement);
+
+            }
+        } catch (IOException e) {
+            Log.e("MainActivity", "error reading assets", e);
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
            Log.e(TAG, "error reading assets", e);
```
Change the first variable of Log to `TAG` and then make that a string constant like "DONATION-BOI"

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java (hunk 11 above)
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]:
```suggestion
```

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+            return city;
+        }
+        private void setState(String st) {
+            state = st;
+        }
+        public String getState() {
+            return state;
+        }
+        private void setZipCode(int zippy) {
+            zipCode = zippy;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: name parameter something else

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+            return streetAddress;
+        }
+        private void setCity(String cit) {
+            city = cit;
+        }
+        public String getCity() {
+            return city;
+        }
+        private void setState(String st) {
+            state = st;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: parameter name is unclear

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+            return longitude;
+        }
+        private void setStreetAddress(String str) {
+            streetAddress = str;
+        }
+        public String getStreetAddress() {
+            return streetAddress;
+        }
+        private void setCity(String cit) {
+            city = cit;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: parameter name is unclear, use `this.city` if you want to name parameter `city`

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+        private String phoneNumber;
+        private String website;
+
+        private ListElement(Object[] args) {
+        }
+        private ListElement() {
+            this(new Object[0]);
+        }
+
+        private void setName(String nam) {
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java]: consider nuking all the getters/setters here and letting Android Studio re-generate them

File: app/src/main/java/com/scrumbums/donationboi/model/util/AccountValidation.java [hunk 17]
```diff
@@ -95,8 +95,7 @@ public static final boolean isValidEmail(String s) {
      * @return If the String is a valid password
      */
     public static final boolean isValidPassword(String s) {
-        return isStringNonEmpty(s) && !containsIllegalCharacters(s)
-                && s.length() >= MINPASSWORDLENGTH && hasMinChars(s);
+        return true;
```
Comment (evan10s) [inline app/src/main/java/com/scrumbums/donationboi/model/util/AccountValidation.java]: revert this change

File: app/src/main/res/layout/activity_main.xml [hunk 18]
```diff
@@ -24,11 +24,13 @@
         android:layout_width="188dp"
         android:layout_height="48dp"
         android:text="@string/login_button"
+
```
Comment (evan10s) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
```

File: app/src/main/res/layout/activity_main.xml [hunk 19]
```diff
@@ -24,11 +24,13 @@
...
         app:layout_constraintBottom_toTopOf="@+id/button_register"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent" />
```
Comment (evan10s) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
```

File: app/src/main/res/layout/activity_main.xml [hunk 20]
```diff
@@ -24,11 +24,13 @@
...
 
     <Button
+
```
Comment (evan10s) [inline app/src/main/res/layout/activity_main.xml]: remove all the weird whitespace

File: app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java (hunk 3 above)
Comment (gibrane) [reply app/src/main/java/com/scrumbums/donationboi/controllers/ListElementPage.java]: will be fixed in update
